### PR TITLE
Don't hardcode array type in MultiContinuousSpace.

### PR DESCRIPTION
**Pull request by aterenin**
Right now this is hardcoded to Array{Float64}, which prevents users using, for example, CuArray{Float32} instead given an appropriate environment.

### CODE REVIEW

**Comment by aterenin** [comment]
Huh, where did that CI failure come from? Tests all passed on my system before I pushed - strange. Will fix this, provided people want the change in this PR.

File: src/spaces/multi_continuous_space.jl [hunk 1]
```diff
@@ -2,24 +2,25 @@ export MultiContinuousSpace
...
-    high::Array{Float64,N}
-    function MultiContinuousSpace(low::Array{Float64}, high::Array{Float64})
-        size(low) == size(high) || throw(ArgumentError("$(size(low)) != $(size(high)), size must match"))
-        all(l < h for (l, h) in zip(
-            low,
-            high,
-        )) || throw(ArgumentError("each element of $low must be less than $high"))
-        new{size(low),ndims(low)}(low, high)
-    end
+struct MultiContinuousSpace{S, N, Fl<:AbstractFloat, AFl<:AbstractArray{Fl,N}} <: AbstractDiscreteSpace
```
Comment (jbrea) [inline src/spaces/multi_continuous_space.jl:5]: Does anything speak against `MultiContinuousSpace{S, N, A <: AbstractArray{<:Number, N}}`?

File: src/spaces/multi_continuous_space.jl [hunk 2]
```diff
@@ -2,24 +2,25 @@ export MultiContinuousSpace
...
+  )) || throw(ArgumentError("each element of $low must be less than $high"))
+  MultiContinuousSpace{size(low),Fl,ndims(low),AFl}(low, high)
+end
+
+MultiContinuousSpace(low, high; float_type = Float64) =
+    MultiContinuousSpace(convert(Array{float_type}, low), convert(Array{float_type}, high))
 
-Base.eltype(::MultiContinuousSpace{S,N}) where {S,N} = Array{Float64,N}
-Base.in(xs, s::MultiContinuousSpace{S,N}) where {S,N} =
+Base.eltype(::MultiContinuousSpace{S,N,Fl}) where {S,Fl,N} = Array{Fl,N}
```
Comment (jbrea) [inline src/spaces/multi_continuous_space.jl:22]: I guess this should then become `Base.eltype(::MultiContinuousSpace{S,N,AF1}) where {S,N,A} = A`.

File: src/spaces/multi_continuous_space.jl [hunk 3]
```diff
@@ -2,24 +2,25 @@ export MultiContinuousSpace
...
+Base.in(xs, s::MultiContinuousSpace) =
```
Comment (jbrea) [inline src/spaces/multi_continuous_space.jl:23]: This doesn't work without `Base.in(xs, s::MultiContinuousSpace{S}) where S =`

**Review by jbrea — COMMENTED** [review]
I like this idea.

**Comment by findmyway** [comment]
The idea sounds great. I'll integrate this feature in the https://github.com/JuliaReinforcementLearning/ReinforcementLearningCore.jl/pull/1

<hr>

Due to the heavy [Wuhan_coronavirus_outbreak](https://en.wikipedia.org/wiki/2019%E2%80%9320_Wuhan_coronavirus_outbreak) , I have plenty of time to work on this project in the next several weeks. Hope we can move forward and make a release soon.

**Comment by findmyway** [comment]
FYI: 

https://github.com/JuliaStats/Distributions.jl/issues/960

**Comment by findmyway** [comment]
Added  here: https://github.com/JuliaReinforcementLearning/ReinforcementLearningCore.jl/pull/2/files/0a3876e21ddee24fb352c6194ec3d86c8a4ff031..fefccb2da9427ddbb97680498fe92e25eefd115f

**Comment by aterenin** [comment]
Sorry for the delay with this - ICML deadline has been squeezing my time. I can get to this PR as soon as that deadline passes, if still needed by then!

**Comment by aterenin** [comment]
It seems this has since been rewritten and moved into ReinforcementLearningCore.

**Pull request closed without merging**